$NetBSD: patch-CorsixTH_CMakeLists.txt,v 1.4 2025/09/27 09:16:12 triaxx Exp $

Fix <app> argument to prepare the bundle installed in ${DESTDIR}.

--- CorsixTH/CMakeLists.txt.orig	2025-08-08 19:15:08.000000000 +0000
+++ CorsixTH/CMakeLists.txt
@@ -24,7 +24,7 @@ elseif(MSVC)
   set(CORSIX_TH_DATADIR CorsixTH)
   set(CORSIX_TH_INTERPRETER_PATH ${CORSIX_TH_INTERPRETER_NAME})
 elseif(APPLE)
-  set(CORSIX_TH_DATADIR CorsixTH.app/Contents/Resources/)
+  set(CORSIX_TH_DATADIR @MACOS_BUNDLE_DIR@/CorsixTH.app/Contents/Resources/)
   set(CORSIX_TH_INTERPRETER_PATH ${CMAKE_INSTALL_PREFIX}/${CORSIX_TH_DATADIR}/${CORSIX_TH_INTERPRETER_NAME})
 else()
   set(CORSIX_TH_DATADIR ${CMAKE_INSTALL_DATADIR}/corsix-th)
@@ -273,7 +273,7 @@ endif()
 if(NOT USE_SOURCE_DATADIRS)
   if(APPLE)
     # Just use the prefix as it's sufficient to just set the prefix to /Applications on Mac.
-    install(TARGETS CorsixTH BUNDLE DESTINATION .)
+    install(TARGETS CorsixTH BUNDLE DESTINATION @MACOS_BUNDLE_DIR@)
   elseif(MSVC)
     install(TARGETS CorsixTH
       RUNTIME DESTINATION ${CORSIX_TH_DATADIR}
@@ -319,7 +319,7 @@ if(NOT USE_SOURCE_DATADIRS)
     install(CODE "
       INCLUDE(BundleUtilities)
       SET(BU_CHMOD_BUNDLE_ITEMS ON)
-      FIXUP_BUNDLE(\"${CMAKE_INSTALL_PREFIX}/CorsixTH.app\" \"\" \"\")
+      FIXUP_BUNDLE(\"@CMAKE_INSTALLED_BUNDLE_APP@\" \"\" \"\")
       ")
     if(WITH_LUAROCKS)
       install(CODE "execute_process(
